lib/Swagger/src/TemplateClassification/Storage/Models/TemplateTypeProvider.php

```diff
@@ -34,222 +42,260 @@
+        return $this->container['provider'];
     }
-  
+
     /**
      * Sets provider
-     * @param string $provider 
+     * @param string $provider
      * @return $this
      */
     public function setProvider($provider)
     {
-        
-        $this->provider = $provider;
+        $this->container['provider'] = $provider;
+
         return $this;
     }
-    
+
     /**
      * Gets origin
      * @return string
      */
     public function getOrigin()
     {
-        return $this->origin;
+        return $this->container['origin'];
     }
-  
+
     /**
      * Sets origin
-     * @param string $origin 
+     * @param string $origin
      * @return $this
      */
     public function setOrigin($origin)
     {
-        
-        $this->origin = $origin;
+        $this->container['origin'] = $origin;
+
         return $this;
     }
-    
+
     /**
      * Gets types
      * @return string[]
      */
     public function getTypes()
     {
-        return $this->types;
+        return $this->container['types'];
     }
-  
+
     /**
      * Sets types
-     * @param string[] $types 
+     * @param string[] $types
      * @return $this
      */
     public function setTypes($types)
     {
-        
-        $this->types = $types;
+        $this->container['types'] = $types;
+
         return $this;
     }
-    
     /**
      * Returns true if offset exists. False otherwise.
-     * @param  integer $offset Offset 
+     * @param  integer $offset Offset
      * @return boolean
      */
     public function offsetExists($offset)
     {
-        return isset($this->$offset);
+        return isset($this->container[$offset]);
     }
-  
+
     /**
      * Gets offset.
-     * @param  integer $offset Offset 
-     * @return mixed 
+     * @param  integer $offset Offset
+     * @return mixed
      */
     public function offsetGet($offset)
     {
-        return $this->$offset;
+        return isset($this->container[$offset]) ? $this->container[$offset] : null;
     }
-  
+
     /**
      * Sets value based on offset.
-     * @param  integer $offset Offset 
+     * @param  integer $offset Offset
      * @param  mixed   $value  Value to be set
      * @return void
      */
     public function offsetSet($offset, $value)
     {
-        $this->$offset = $value;
+        if (is_null($offset)) {
+            $this->container[] = $value;
+        } else {
+            $this->container[$offset] = $value;
+        }
     }
-  
+
     /**
      * Unsets offset.
-     * @param  integer $offset Offset 
+     * @param  integer $offset Offset
      * @return void
      */
     public function offsetUnset($offset)
     {
-        unset($this->$offset);
+        unset($this->container[$offset]);
     }
-  
+
     /**
      * Gets the string presentation of the object
      * @return string
      */
     public function __toString()
     {
-        if (defined('JSON_PRETTY_PRINT')) {
+        if (defined('JSON_PRETTY_PRINT')) { // use JSON pretty print
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)